qt/python/instrumentview/instrumentview/Peaks/WorkspaceDetectorPeaks.py

```
@@ -12,50 +12,42 @@
+            Peak(det_id, peak_idx, hkl, tof, dspacing, wavelength, 2 * np.pi / dspacing)
+            for (det_id, peak_idx, hkl, tof, dspacing, wavelength) in zip(
+                detector_ids, range(len(tofs)), hkls, tofs, dspacings, wavelengths, strict=True
             )
         ]
         # groupby groups consecutive matches, so must be sorted
-        peaks.sort(key=lambda x: x.spectrum_no)
-        for spec_no, peaks_for_spec in groupby(peaks, lambda x: x.spectrum_no):
-            if spec_no in all_spectrum_nos:
-                self.detector_peaks.append(DetectorPeaks(list(peaks_for_spec)))
+        peaks.sort(key=lambda x: x.detector_id)
+        for spec_no, peaks_for_spec in groupby(peaks, lambda x: x.detector_id):
```

Ruff (0.15.12)

[warning] Loop control variable `spec_no` not used within loop body

Rename unused `spec_no` to `_spec_no`

(B007)